src/modules/account/RadixTransferAccountSystem.ts

```diff
@@ -109,57 +130,116 @@ export default class RadixTransferAccountSystem implements RadixAccountSystem {
+                    if (!state.spentConsumables.has(hid)) {
+                        state.unspentConsumables.set(hid, particle)
                     }
                 }
 
-                if (!(tokenClassReference.toString() in transaction.balance)) {
-                    transaction.balance[tokenClassReference.toString()] = new BN(0)
+                if (!(tokenClassReference.toString() in currentBalance)) {
+                    currentBalance[tokenClassReference.toString()] = new BN(0)
                 }
-                transaction.balance[tokenClassReference.toString()].iadd(quantity)
+                currentBalance[tokenClassReference.toString()].iadd(quantity)
             } else {
-                transaction.participants[particle.getOwner().toString()] = particle.getOwner()
+                currentParticipants[particle.getOwner().toString()] = particle.getOwner()
             }
         }
 
         // Not a transfer
-        if (Object.keys(transaction.balance).length === 0) {
+        if (Object.keys(currentBalance).length === 0) {
             return
         }
 
 
-        const numberOfParticipants = Object.keys(transaction.participants).length
+        const numberOfParticipants = Object.keys(currentParticipants).length
         if (numberOfParticipants > 2) {
             throw new Error(`Invalid number of transaction participants = ${numberOfParticipants}`)
         }
 
         // Update balance
-        for (const tokenId in transaction.balance) {
-            // Load tokenclass from network
-            // const tokenClass = await radixTokenManager.getTokenClass(tokenId)
-
-            if (!(tokenId in this.balance) || !this.balance[tokenId]) {
-                this.balance[tokenId] = new BN(0)
+        for (const tokenId in currentBalance) {
+            if (!(tokenId in state.balance) || !state.balance[tokenId]) {
+                state.balance[tokenId] = new BN(0)
             }
 
-            this.balance[tokenId].iadd(transaction.balance[tokenId])
+            state.balance[tokenId].iadd(currentBalance[tokenId])
 
             // Token units
-            transaction.tokenUnitsBalance[tokenId] = RadixTokenDefinition.fromSubunitsToDecimal(transaction.balance[tokenId])
+            currentTokenUnitsBalance[tokenId] = RadixTokenDefinition.fromSubunitsToDecimal(currentBalance[tokenId])
 
-            if (!(tokenId in this.tokenUnitsBalance) || !this.balance[tokenId]) {
-                this.tokenUnitsBalance[tokenId] = new Decimal(0)
+            if (!(tokenId in state.tokenUnitsBalance) || !state.balance[tokenId]) {
+                state.tokenUnitsBalance[tokenId] = new Decimal(0)
             }
 
-            this.tokenUnitsBalance[tokenId] = this.tokenUnitsBalance[tokenId].add(transaction.tokenUnitsBalance[tokenId])
+            state.tokenUnitsBalance[tokenId] = state.tokenUnitsBalance[tokenId].add(currentTokenUnitsBalance[tokenId])
         }
 
+        return {
+            currentBalance,
```

Comment:
Not sure about naming things in the return object "currentXyz". Current as opposed to what?

Reply:
removed